### PR TITLE
Show last cron messages on monitor page

### PR DESCRIPTION
**Pull request by zeripath**
As discussed on #19221 we should store the results of the last task message on the
crontask and show them on the monitor page.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -2815,6 +2815,7 @@ monitor.process = Running Processes
 monitor.desc = Description
 monitor.start = Start Time
 monitor.execute_time = Execution Time
+monitor.last_execution_result = Success?
```
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: The text could be `Result`?

`Success?` reads a little strange on UI in my mind.

File: services/cron/setting.go [hunk 2]
```diff
@@ -70,19 +68,19 @@ func (b *BaseConfig) DoNoticeOnSuccess() bool {
...
 	if len(args) > 0 {
 		realArgs = append(realArgs, args...)
 	}
-	if doer == nil || (doer.ID == -1 && doer.Name == "(Cron)") {
-		return i18n.Tr("en-US", "admin.dashboard.cron."+status, realArgs...)
+	if doer == "" {
+		return i18n.Tr(locale, "admin.dashboard.cron."+status, realArgs...)
 	}
-	return i18n.Tr("en-US", "admin.dashboard.task."+status, realArgs...)
+	return i18n.Tr(locale, "admin.dashboard.task."+status, realArgs...)
```
Comment (wxiaoguang) [inline services/cron/setting.go:86]: Before, there was a status `aborted`, now, a new status `cancelled`.

The i18n keys would mismatch for these cases.

And I think we need some consts for these statuses. And we need some comments to remember that these status names are used for i18n keys.

Comment (zeripath) [reply services/cron/setting.go:86]: If you check the locale files you'll see that I've changed to cancelled to match the locale files ...

Comment (wxiaoguang) [reply services/cron/setting.go:86]: Hmm ... strange, GitHub diff doesn't show the change from `dashboard.cron.aborted` to `dashboard.cron.cancelled`.

Just viewed the full file `options/locale/locale_en-US.ini`, `dashboard.cron.cancelled` is indeed there ..... maybe the old code's `aborted` was incorrect before.

No problem from my side, although I think a comment would help to clarify the relationship between these status names and i18n keys.

**Review by wxiaoguang — APPROVED** [review]

File: options/locale/locale_en-US.ini (hunk 1 above)
Comment (zeripath) [reply options/locale/locale_en-US.ini]:
```suggestion
monitor.last_execution_result = Result
```

File: services/cron/setting.go (hunk 2 above)
Comment (zeripath) [reply services/cron/setting.go:86]: The old code "aborted" was indeed incorrect

File: services/cron/setting.go [hunk 3]
```diff
@@ -70,19 +68,19 @@ func (b *BaseConfig) DoNoticeOnSuccess() bool {
 }
 
 // FormatMessage returns a message for the task
-func (b *BaseConfig) FormatMessage(name, status string, doer *user_model.User, args ...interface{}) string {
+func (b *BaseConfig) FormatMessage(locale, name, status, doer string, args ...interface{}) string {
```
Comment (zeripath) [inline services/cron/setting.go:72]:
```suggestion
// Please note the `status` string will be concatenated with `admin.dashboard.cron.` and `admin.dashboard.task.` to provide locale messages. Similarly `name` will be composed with `admin.dashboard.` to provide the locale name for the task.
func (b *BaseConfig) FormatMessage(locale, name, status, doer string, args ...interface{}) string {
```

File: services/cron/setting.go (hunk 2 above)
Comment (zeripath) [reply services/cron/setting.go:86]: @wxiaoguang I've updated the comment of the function to add more information.

**Review by Gusted — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**